bgpd/bgpd.c

```diff
@@ -7681,6 +7681,35 @@ bgp_if_finish (struct bgp *bgp)
 
 extern void bgp_snmp_init (void);
 
+static void
+bgp_viewvrf_autocomplete (vector comps, struct cmd_token *token)
+{
+  struct vrf *vrf = NULL;
+  struct listnode *next;
+  struct bgp *bgp;
+
+  RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name)
+    {
+      if (vrf->vrf_id != VRF_DEFAULT)
+        vector_set (comps, XSTRDUP (MTYPE_COMPLETION, vrf->name));
+    }
+
+  for (ALL_LIST_ELEMENTS_RO (bm->bgp, next, bgp))
+    {
+      if (bgp->inst_type != BGP_INSTANCE_TYPE_VIEW)
+        continue;
+
+      vector_set (comps, XSTRDUP (MTYPE_COMPLETION, bgp->name));
+    }
+}
+
+static const struct cmd_variable_handler bgp_viewvrf_var_handlers[] = {
+  {
+    .tokenname = "VIEWVRFNAME",
+    .completions = bgp_viewvrf_autocomplete
+  },
```

Comment:
you need a null terminator sentinel element here (`{ .completions = NULL }`)
(i should document that more explicitly i guess?)